src/connection/eagerlyConnect.ts

```diff
@@ -0,0 +1,36 @@
+import { Connector } from '@web3-react/types'
+
+import { getConnection, gnosisSafeConnection, networkConnection } from './index'
+import { ConnectionType, selectedWalletKey, toConnectionType } from './types'
+
+async function connect(connector: Connector, type: ConnectionType) {
+  performance.mark(`web3:connect:${type}:start`)
+  try {
+    if (connector.connectEagerly) {
```

Comment:
i **think** could be fancy with `await (connector.connectEagerly?.() ?? connector.activate())`

Reply:
connectEagerly returns *something* asynchronously, and I'm not sure what it is. But I *think* this would work.
Still, KISS.